src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1964,5 +1965,13 @@ internal static int TypeToIndex(this TypeSymbol type)
                 default: return -1;
             }
         }
+
+        internal static void CheckRecordLangVersion(this TypeSymbol type, DiagnosticBag diagnostics, TypeSyntax syntax, Location location)
```

Comment:
>Location location [](start = 120, length = 17)

It looks like all the consumers are passing location of the previous parameter. Consider removing this parameter and requesting location inside this helper when we actually need it.